core/libcamera_app.hpp

```diff
@@ -104,22 +105,20 @@ class LibcameraApp
 	Msg Wait();
 	void PostMessage(MsgType &t, MsgPayload &p);
 
-	Stream *GetStream(std::string const &name, unsigned int *w = nullptr, unsigned int *h = nullptr,
-					  unsigned int *stride = nullptr) const;
-	Stream *ViewfinderStream(unsigned int *w = nullptr, unsigned int *h = nullptr,
-							 unsigned int *stride = nullptr) const;
-	Stream *StillStream(unsigned int *w = nullptr, unsigned int *h = nullptr, unsigned int *stride = nullptr) const;
-	Stream *RawStream(unsigned int *w = nullptr, unsigned int *h = nullptr, unsigned int *stride = nullptr) const;
-	Stream *VideoStream(unsigned int *w = nullptr, unsigned int *h = nullptr, unsigned int *stride = nullptr) const;
-	Stream *LoresStream(unsigned int *w = nullptr, unsigned int *h = nullptr, unsigned int *stride = nullptr) const;
+	Stream *GetStream(std::string const &name, StreamInfo *info = nullptr) const;
+	Stream *ViewfinderStream(StreamInfo *info = nullptr) const;
+	Stream *StillStream(StreamInfo *info = nullptr) const;
+	Stream *RawStream(StreamInfo *info = nullptr) const;
+	Stream *VideoStream(StreamInfo *info = nullptr) const;
+	Stream *LoresStream(StreamInfo *info = nullptr) const;
 	Stream *GetMainStream() const;
 
 	std::vector<libcamera::Span<uint8_t>> Mmap(FrameBuffer *buffer) const;
 
 	void ShowPreview(CompletedRequestPtr &completed_request, Stream *stream);
 
 	void SetControls(ControlList &controls);
-	void StreamDimensions(Stream const *stream, unsigned int *w, unsigned int *h, unsigned int *stride) const;
+	void GetStreamInfo(Stream const *stream, StreamInfo *info) const;
```

Comment:
I wonder very slightly about simply making this return-by-value now, but honestly, it's so trivial I don't really mind...

Reply:
Good idea.